runtime/queries/c/highlights.scm

```diff
@@ -76,17 +152,31 @@
 (function_declarator
   declarator: (identifier) @function)
 (preproc_function_def
-  name: (identifier) @function.special)
+  name: (identifier) @function.macro)
 
-(field_identifier) @variable.other.member
-(statement_identifier) @label
-(type_identifier) @type
-(primitive_type) @type
-(sized_type_specifier) @type
+(comment) @comment
 
-((identifier) @constant
- (#match? @constant "^[A-Z][A-Z\\d_]*$"))
+;; Parameters
+(parameter_declaration
+  declarator: (identifier) @variable.parameter)
+(parameter_declaration
+  declarator: (pointer_declarator) @variable.parameter)
+(preproc_params (identifier) @variable.parameter)
 
-(identifier) @variable
+[
+  "__attribute__"
+  "__cdecl"
+  "__clrcall"
+  "__stdcall"
+  "__fastcall"
+  "__thiscall"
+  "__vectorcall"
+  "_unaligned"
+  "__unaligned"
+  "__declspec"
+  (attribute_declaration)
+] @attribute
 
-(comment) @comment
+(field_identifier) @variable.other.member
+
+(ERROR) @error
```

Comment:
I've seen this added more and more, I've had this added to highlights automatically before: https://github.com/helix-editor/helix/blob/c2e41082e4b2e0e1dba6a980a7f09ccaad88cbd0/helix-core/src/syntax.rs#L359-L360

I think we should remove this everywhere, use the snippet above and make it a config option.

Reply:
Will it cause the query to fail if an `ERROR` node isn't defined for the grammar?

Reply:
The `ERROR` node is a meta node automatically inserted by tree-sitter on a parsing error.